File: Synchro.Frontend/src/models/question.ts
export interface QuestionDataDto {
  totalQuestionsCount?: number;
  answeredQuestionsCount?: number;
  correctlyAnsweredQuestionsCount?: number;
  correctlyAnsweredQuestionsPercentage?: number;
  questions?: QuestionViewDto[];
}

export interface QuestionViewDto {
  isCorrect: boolean;
  image?: string;
  text: string;
  topics?: string[];
  userAnswer: string;
  correctAnswer: string;
}

export interface QuestionDto {
  id: string;
  text: string;
  options: string[];
  image?: string;
  correctAnswer: string;
  topics?: string[];
  createdBy: string;
  createdAt: string;
  lastModifiedBy?: string;
  lastModifiedAt?: string;
}

File: Core/Domain/Entities/Question.cs
using Core.Interfaces;
using System.ComponentModel.DataAnnotations;

namespace Core.Domain.Entities
{
    public class Question : IBaseEntity
    {
        public Guid Id { get; set; }
        public string? Text { get; set; }
        public required IList<string> Options { get; set; }
        public string? Image { get; set; }
        public IList<string>? Topics { get; set; }
        public required string CorrectAnswer { get; set; }
        public ICollection<Answer> Answers { get; set; } = [];
        public string CreatedBy { get; set; }
        public string? LastModifiedBy { get; set; }
        public DateTimeOffset CreatedAt { get; set; }
        public DateTimeOffset? LastModifiedAt { get; set; }
        [Timestamp]
        public byte[] RowVersion { get; set; }
    }
}
